### PR TITLE
Add better support for isotopic Hs to removeHs() and addHs()

### PR DESCRIPTION
**Pull request by ptosco**
@greglandrum This is the functionality that we chatted about on Slack a couple of days ago.
I added a `removeAndTrackIsotopes` flag to `RemoveHsParameters` which, in addition to triggering removal of isotopic Hs will track them in a private `_isotopicHs` atom property such that `addHs()` can then put them back as isotopes in the right places.
I chose to use a tuple-formatted string property, as using a `std::vector` or `std::deque` C++ property would make it hardly accessible from Python without adding a specific `GetProp` flavor. Given the conversion from/to string only happens once I think the performance hit is negligible and allows to simply `eval` (or `ast.literal_eval`) the tuple string, which seems simpler to me.

### CODE REVIEW

**Comment by greglandrum** [comment]
@ptosco : I have a reflexive adversion to anything that requires eval.
Any reason to not just format this by doing a space-delimited string? Then the Python parsing becomes: `[int(x) for x in propval.split(' ')]`, which is essentially just as simple and makes it much more clear what the property value actually looks like

**Comment by greglandrum** [comment]
Also: it looks like the change to FindCairo.cmake is unconnected to this PR. Did that sneak in accidentally?

**Comment by ptosco** [comment]
@greglandrum yes, `FindCairo.cmake` sneaked in accidentally.
BTW, is it only me having issues with `libXau` not being found when building on CentOS 7?
I am seeing this consistently lately.
I’ll revert this and address your other comment. Thanks!

**Comment by greglandrum** [comment]
@ptosco you might want to hold off a bit. I'm finishing up my review and it will like include a PR against your fork with some suggested changes

**Comment by greglandrum** [comment]
@ptosco: I'm going back through older PRs. I think this one is waiting for you to look at the PR I did against your branch.

**Comment by ptosco** [comment]
@greglandrum Yep, it's on my to-do list.

**Comment by ptosco** [comment]
@greglandrum I merged your PR. The use case that I had in mind was about changes in hybridization state or substitution, but your code will stop adding isotopes when they do not fit anymore anyway. Having though about it, there is no point in keeping records of isotopes that have not been added yet, as it most likely does not makes sense to add them later on if hybridization changes again.
I removed the check for emptyness of the `std::vector` returned by `getIsoMap()` as the returned map cannot have keys with corresponding empty vector.
This is good to go for me once tests have passed. Thanks!

File: Code/cmake/Modules/FindCairo.cmake
Comment (greglandrum) [inline Code/cmake/Modules/FindCairo.cmake:56]: The changes to this file should not be part of this PR

**Review by greglandrum — CHANGES_REQUESTED** [review]
Once the FindCairo.cmake changes are removed, I think this is ready to go

File: Code/GraphMol/Wrap/rough_test.py
Comment (greglandrum) [inline Code/GraphMol/Wrap/rough_test.py:5923]: a philosophical comment, not really suggesting a change:
Though this construction directly tests what you're trying to test, it's reasonably non-trivial to read and figure out what it's doing. In order to lessen the cognitive load on the person reading through the tests, I'd probably just use MolToSmiles() to test that the right thing has happened here.

File: Code/GraphMol/molopstest.cpp
Comment (greglandrum) [inline Code/GraphMol/molopstest.cpp:7659]: again, just a comment, not a request for anything to be changed.
This test is almost as complex as the code it's testing. I think it's a good idea to keep the testing code itself as simple as possible (it may be that this is true here) so that we don't end up in the awkward situation of having to debug our tests.

File: Code/cmake/Modules/FindCairo.cmake
Comment (ptosco) [reply Code/cmake/Modules/FindCairo.cmake]: I thought I had done that, but obviously I hadn't - sorry.

File: Code/GraphMol/Wrap/rough_test.py
Comment (ptosco) [reply Code/GraphMol/Wrap/rough_test.py:5923]: Duly noted. I have done much worse than that, though, as you point out later on : )

File: Code/GraphMol/molopstest.cpp
Comment (ptosco) [reply Code/GraphMol/molopstest.cpp:7659]: In general, I agree with you. In this specific case, I felt that adding a more complex test case with comments would remind me of what I wanted to achieve with the code better than a test case that just checks the outcomes. I promise not to do this too often : )

**Review by greglandrum — APPROVED** [review]

**Comment by greglandrum** [comment]
LGTM
will merge after the CI builds are done

**Comment by ptosco** [comment]
@greglandrum done. Thanks for all your feedback!

**Pull request merged**
